lib/src/pigeons/map_interfaces.dart

```diff
@@ -1289,6 +1307,257 @@ class QueriedFeature {
   }
 }
 
+/// Identifies a feature in a featureset.
+///
+/// Knowing the feature identifier allows to set the feature states to a particular feature, see ``MapboxMap/setFeatureState(featureset:featureId:state:callback:)``.
+///
+/// In a featureset a feature can come from different underlying sources. In that case their IDs are not guaranteed to be unique in the featureset.
+/// The ``FeaturesetFeatureId/namespace`` is used to disambiguate from which source the feature is coming.
+///
+/// - Warning: There is no guarantee of identifier persistency. This depends on the underlying source of the features and may vary from style to style.
+/// If you want to store the identifiers persistently, please make sure that the style or source provides this guarantee.
+class FeaturesetFeatureId {
+  FeaturesetFeatureId({
+    required this.id,
+    this.namespace,
+  });
+
+  /// A feature id coming from the feature itself.exp
+  String id;
+
+  /// A namespace of the feature
+  String? namespace;
+
+  Object encode() {
+    return <Object?>[
+      id,
+      namespace,
+    ];
+  }
+
+  static FeaturesetFeatureId decode(Object result) {
+    result as List<Object?>;
+    return FeaturesetFeatureId(
+      id: result[0]! as String,
+      namespace: result[1] as String?,
+    );
+  }
+}
+
+/// Wraps a FeatureState map
+class FeatureState {
+  FeatureState({
+    required this.map,
+  });
+
+  Map<String, Object?> map;
+
+  Object encode() {
+    return <Object?>[
+      map,
+    ];
+  }
+
+  static FeatureState decode(Object result) {
+    result as List<Object?>;
+    return FeatureState(
+      map: (result[0] as Map<Object?, Object?>?)!.cast<String, Object?>(),
+    );
+  }
+}
+
+/// An interaction that can be added to the map.
+///
+/// To create an interaction use ``TapInteraction`` and ``LongClickInteraction`` implementations.
+///
+/// See also: ``MapboxMap/addInteraction``.
+class Interaction {
+  Interaction({
+    required this.featuresetDescriptor,
+    required this.interactionType,
+    required this.stopPropagation,
+    this.filter,
+    this.radius,
+  });
+
+  /// The featureset descriptor that specifies the featureset to be included in the interaction.
+  FeaturesetDescriptor featuresetDescriptor;
+
+  /// The type of interaction, either tap or longTap
+  InteractionType interactionType;
+
+  /// Whether to stop the propagation of the interaction to the map. Defaults to true.
+  bool stopPropagation;
```

Comment:
Not that I've included this here rather than in the callback

Reply:
Today in offline discussion with @evil159 I realized that we cannot implement it at all due to async communication dart<->platform. So this configuration totally makes sense, resolving this convo 👍